### PR TITLE
Added canonical gitignores for Rust, CLion and Bazel from gitignore.io

### PR DESCRIPTION
**Pull request by aalmos**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,42 @@
...
-# Rust artifacts.
-/**/target
+### Rust ###
+# Generated by Cargo
+# will have compiled files and executables
+/**/target/
+
+# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
+# More information here https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html
+Cargo.lock
```
Comment (tiziano88) [inline .gitignore]: We should not ignore Cargo.lock, it contains the list of specific dependencies used to reproducibly build our examples. I think the place you got this .gitignore from assumes that you are developing a Rust library, in which case you would typically not check in Cargo.lock, just Cargo.toml

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,42 @@
...
+
+# These are backup files generated by rustfmt
+**/*.rs.bk
+
+### CLion ###
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio and WebStorm
+# Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
+
+# User-specific stuff
+.idea/**/workspace.xml
```
Comment (tiziano88) [inline .gitignore]: Can we just ignore `.idea` entirely? Or do you think there is a reasonable use case for checking in some of this stuff?

Comment (aalmos) [reply .gitignore]: I think once we can make the project load in CLion properly it'd be reasonable to submit the .idea files. Anyways CLion is a Google supported IDE for C++, so it'd make sense.

File: .gitignore (hunk 1 above)
Comment (aalmos) [reply .gitignore]: OK, I removed that part.

File: .gitignore (hunk 2 above)
Comment (tiziano88) [reply .gitignore]: In my experience it is rarely a good idea to check in specific IDE or editor configuration files, as developers have different needs / preferences. That said, I do not know anything about CLion so I am happy to follow your advice is you think that whatever is going to eventually be checked in is sufficiently non-controversial that everyone using CLion would be happy with.

Comment (aalmos) [reply .gitignore]: SGTM.

**Pull request merged**